DataDog/datadog-operator
# [defaults] Fix cluster agent token autogeneration

### What does this PR do?

Currently, when a cluster agent token is not specified, it is not autogenerated properly and agents cannot connect to the cluster agent. This PR fixes the issue.

When a token is not specified, this PR autogenerates one and stores it in the "DefaultOverride" attribute of the spec.

### Additional Notes

When the secret that contains the token changes, the cluster agent and the agents are not restarted, so they keep using the old token. This issue is not fixed by this PR yet.

### Describe your test plan

- Deploy a datadog agent without specifying a token in the spec (`credentials.token`). Check the status of the agents (`agent status`) and verify that they can connect with the cluster agent. Also check that when the datadog agent is deleted, the secret with the token (has the same name as the datadog agent object) is also deleted. You can also check that in the status of the datadog agent, `defaultOverride.credentials.token` matches the token stored in the secret.
- Deploy a datadog agent specifying a token in the spec (`credentials.token`) and check the same except the `defaultOverride` part, which doesn't apply in this case.
